ElectronicObserverTypes/Attacks/Extensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace ElectronicObserverTypes.Attacks;
+
+public static class Extensions
+{
+	public static bool IsSpecialAttack(this DayAttackKind dayAttack) => dayAttack is
+		DayAttackKind.SpecialNelson or
+		DayAttackKind.SpecialNagato or
+		DayAttackKind.SpecialMutsu or
+		DayAttackKind.SpecialColorado or
+		DayAttackKind.SpecialKongo or
+		DayAttackKind.SpecialSubmarineTender23 or
+		DayAttackKind.SpecialSubmarineTender34 or
+		DayAttackKind.SpecialSubmarineTender24 or
+		DayAttackKind.SpecialYamato2Ships or
+		DayAttackKind.SpecialYamato3Ships;
+
+	public static bool IsSpecialAttack(this NightAttackKind nightAttack) => nightAttack is
+		NightAttackKind.CutinZuiun or
+		NightAttackKind.SpecialNelson or
+		NightAttackKind.SpecialNagato or
+		NightAttackKind.SpecialMutsu or
+		NightAttackKind.SpecialColorado or
+		NightAttackKind.SpecialKongou or
+		NightAttackKind.SpecialSubmarineTender23 or
+		NightAttackKind.SpecialSubmarineTender34 or
+		NightAttackKind.SpecialSubmarineTender24 or
+		NightAttackKind.SpecialYamato2Ships or
+		NightAttackKind.SpecialYamato3Ships;
+
+	public static List<int> SpecialAttackParticipationIndexes(this DayAttackKind? dayAttack)
+		=> dayAttack switch
+		{
+			{ } atk => atk switch
+			{
+				DayAttackKind.SpecialSubmarineTender23 or
+				DayAttackKind.SpecialSubmarineTender34 or
+				DayAttackKind.SpecialSubmarineTender24 => SpecialAttackIndexes(atk)
+					.Append(0)
+					.Distinct()
+					.ToList(),
+
+				_ => SpecialAttackIndexes(atk)
+					.Distinct()
+					.ToList()
+			},
+
+			_ => [],
+		};
+
+	public static List<int> SpecialAttackParticipationIndexes(this NightAttackKind? nightAttack)
+		=> nightAttack switch
+		{
+			{ } atk => atk switch
+			{
+				NightAttackKind.SpecialSubmarineTender23 or
+				NightAttackKind.SpecialSubmarineTender34 or
+				NightAttackKind.SpecialSubmarineTender24 => SpecialAttackIndexes(atk)
+					.Append(0)
+					.Distinct()
+					.ToList(),
+
+				_ => SpecialAttackIndexes(atk)
+					.Distinct()
+					.ToList()
+			},
+
+			_ => [],
+		};
+
+	public static List<int> SpecialAttackIndexes(this DayAttackKind dayAttack)
+		=> dayAttack switch
+		{
+			DayAttackKind.SpecialNelson => [0, 2, 4],
+
+			DayAttackKind.SpecialNagato or
+			DayAttackKind.SpecialMutsu or
+			DayAttackKind.SpecialYamato2Ships => [0, 0, 1],
+
+			DayAttackKind.SpecialColorado or
+			DayAttackKind.SpecialYamato3Ships => [0, 1, 2],
+
+			DayAttackKind.SpecialSubmarineTender23 => [1, 2, 1, 2],
```

Comment:
I'd expect the attack indexes to be 1, 1, 2, 2 for the case where 4 attacks happen, but I can't find any source for that, so I have no idea what's correct.

Reply:
The sample I got yesterday was 1, 2, 1, 2

Reply:
Hmm, how did you come to that conclusion? From the pics in discord, I can't tell.

Reply:
After checking game's code : 
For every damage, if damage index % 2 == 0, the torpedo comes from first sub, else it comes from second sub
We can assume the logic is the same for damage calculation, but it does require testing since we couldn't find any source on that